server/auth/sso/sso.go

```diff
@@ -280,6 +293,18 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+	if s.filterGroupsRegex != nil && len(s.filterGroupsRegex) > 0 {
+		var filteredGroups []string
```

Comment:
it would be great if the provider could do this logic so that the response from it is not so large, but it does not seem there is a uniform provider interface for that.

Azure AD implements [group filtering per app](https://learn.microsoft.com/en-us/azure/active-directory/hybrid/connect/how-to-connect-fed-group-claims#group-filtering)
LDAP more generally has [group search](https://dexidp.io/docs/connectors/ldap/).

There is a related Dex issue on this (https://github.com/dexidp/dex/issues/1476) and generalizing it was basically closed out due to provider-specific nuances.

So I think there may be no way around filtering in the Server code